raft/src/test/java/org/apache/kafka/raft/MockRequestPurgatory.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.raft;
+
+import org.apache.kafka.common.errors.TimeoutException;
+import org.apache.kafka.common.utils.MockTime;
+
+import java.util.NavigableSet;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class MockRequestPurgatory<T> implements RequestPurgatory<T>, MockTime.Listener {
+    private final AtomicLong idGenerator = new AtomicLong(1L);
+    private final MockTime time;
+    private final NavigableSet<DelayedOperation> delayedOperations = new TreeSet<>();
+
+    public MockRequestPurgatory(MockTime time) {
+        this.time = time;
+        time.addListener(this);
+    }
+
+    @Override
+    public void await(
```

Comment:
The mock purgatory implementation is a bit different from the scala one as the `await()` call doesn't check other buffered delayed operation. Do we want to make them consistent?

Reply:
I don't think we check other operations here either, do we? I followed the path inside `tryCompleteElseWatch` and it looks like we only attempt completion of the new operation.